flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/UserDefinedFunction.scala

```diff
@@ -24,4 +24,19 @@ package org.apache.flink.table.functions
   * User-defined functions must have a default constructor and must be instantiable during runtime.
   */
 trait UserDefinedFunction {
```

Comment:
Did you run your code in Java 7 or Java 8? Java 8 introduces default implementation for interfaces, but Java 7 could cause problems. I think it won't hurt if we convert it into an `abstract class`.